### PR TITLE
Add hook point for `alter table set distributed by`

### PR DESCRIPTION
**Pull request by zhangteng5513**
Add hook point for `alter stable set distributed by` so that we can get its query text and its progress in instrument in the hook function.

### CODE REVIEW

**Comment by hlinnaka** [comment]
typo in title

File: src/backend/commands/tablecmds.c [hunk 1]
```diff
@@ -14123,10 +14125,9 @@ build_ctas_with_dist(Relation rel, DistributedBy *dist_clause,
 	dest = CreateDestReceiver(DestIntoRel);
 
 	/* Create a QueryDesc requesting no output */
-	queryDesc = CreateQueryDesc(stmt, pstrdup("(internal SELECT INTO query)"),
+	queryDesc = CreateQueryDesc(stmt, debug_query_string,
```
Comment (asimrp) [inline src/backend/commands/tablecmds.c]: Is this change intentional?  Not that I'm seeing anything wrong here, but seems unusual.  Most callers of `CreateQueryDesc` pass a string in current memory context (using `pstrdup()`) as `sourceText`.  `debug_query_string` seemed somewhat different, it points to a libpq buffer.

**Comment by asimrp** [comment]
FYI: the PR pipeline failure is a known issue (`udf_exception_blocks_panic_scenarios`), actively being worked on.

File: src/backend/commands/tablecmds.c (hunk 1 above)
Comment (zhangteng5513) [reply src/backend/commands/tablecmds.c]: `CreateQueryDesc ` will call `pstrdup` before set the `sourceText `, so maybe it is unnecessary to call `pstrdup` here.
